tests/QueueableActionTest.php

```diff
@@ -125,4 +127,18 @@ public function an_action_can_have_custom_job_tags()
             return $action->tags() === ['custom_tag', 'tagged_action'];
         });
     }
+
+    /** @test */
+    public function an_action_can_have_a_custom_failed_callback()
+    {
+        Queue::fake();
+
+        $action = new ActionWithFailedMethod();
+
+        $action->onQueue()->execute();
+
+        Queue::assertPushed(ActionJob::class, function ($action) {
+            return $action->failed(new Exception('foo')) === 'foo';
```

Comment:
Shouldn't we assert that the `failed` method was actually called when the action failed to execute?

Reply:
I was trying to find the relevant Laravel code that calls the `failed()` method, but didn't succeed in that. So I decided to just go with this and sort of assume that Laravel correctly calls `failed()` on `ActionJob` as it's a normal queueable job.

If you want though, I can try to find a way to get the job to fail and we can set some global variable from the `failed()` method and then check for it in the test.

Reply:
I think we need to be sure the method is actually called, not sure how to test it though…

Reply:
If we're using the sync driver, I think throwing an exception from handle, catching it in the test and asserting that failed() was executed (I.e. set some global variable) should work. Will test that later today

Reply:
@brendt Added that test now. See latest commit. Ready to be merged?

Reply:
Nice! Thanks